src/main/java/seedu/address/ui/WarningWindow.java

```diff
@@ -0,0 +1,106 @@
+package seedu.address.ui;
+
+import java.util.logging.Logger;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+import seedu.address.commons.core.LogsCenter;
+
+/**
+ * This class is the controller for the warning window pop-up.
+ */
+public class WarningWindow extends UiPart<Stage> {
+
+    private static final Logger logger = LogsCenter.getLogger(WarningWindow.class);
+    private static final String FXML = "WarningWindow.fxml";
+
+    @FXML
+    private Label warningMessage;
+
+    private boolean isOkClicked = false;
+
+    public WarningWindow(Stage root) {
+        super(FXML, root);
+    }
+
+    /**
+     * Creates a new WarningWindow.
+     */
+    public WarningWindow() {
+        this(new Stage());
+    }
+
+    public void setMessage(String message) {
+        warningMessage.setText(message);
+    }
+
+    public boolean isOkClicked() {
+        return isOkClicked;
+    }
+
+    @FXML
+    private void okClicked() {
+        isOkClicked = true;
+        hide();
+    }
+
+    @FXML
+    private void cancelClicked() {
+        hide();
+    }
+
+    /**
+     * Shows the warning window.
+     * @throws IllegalStateException
+     *     <ul>
+     *         <li>
+     *             if this method is called on a thread other than the JavaFX Application Thread.
+     *         </li>
+     *         <li>
+     *             if this method is called during animation or layout processing.
+     *         </li>
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        logger.fine("Showing warning page about the application.");
```

Comment:
There is a bug here.
This line is not logged into the logger.

Reply:
This method isn't actually called, showAndWait() is called instead. Maybe I should remove this method from the code?